app/uk/gov/hmrc/helptosavefrontend/controllers/AccessAccountController.scala

```diff
@@ -72,6 +73,21 @@ class AccessAccountController @Inject() (
             e => {
               logger.warn(s"error retrieving Account details from NS&I, error = $e", htsContext.nino)
             }, {account =>
+              val accountEndDate: Future[Option[LocalDate]] =
```

Comment:
Sorry, but I think this behaves badly in a number of ways.  It is difficulty to test realiably, because the update occurs asynchronously to redirect, so there is no way of being sure when it has completed other than by the logging.  Also call for isClosed occurs asynchronously too, so there is no guarantee of the ordering and so of the affect.

There pattern of code already present for isClosed was not very good, I think.  By then following it the problems have been compounded.

Reply:
It did feel odd that the redirect was not connected to any logic and that the connector results were mapped to loggers, but yes, I followed the format of what was already there. I understand the issue now.